File: frontend/src/types/workRecommendation.ts
/** Stage 11 recommended-work lifecycle (Stage 11B.1/11B.2 backend contract).
 *
 * `status` (PENDING/ACCEPTED/DISMISSED) is an owner lifecycle decision and
 * `is_active` (the underlying Risk/Finding source still being confirmed) is
 * an independent dimension — a recommendation can be ACCEPTED and later
 * become inactive, or DISMISSED while its source is still active. Never
 * collapse the two into one label.
 */
import { SurfacePriceItemSummaryRead } from './workPlan';

export type WorkRecommendationTriggerTypeValue = 'RISK_RULE' | 'FINDING';

export type WorkRecommendationTargetKindValue = 'WALL' | 'FLOOR' | 'CEILING' | 'ROOM';

export type WorkRecommendationStatusValue = 'PENDING' | 'ACCEPTED' | 'DISMISSED';

export type WorkRecommendationActivityValue = 'active' | 'resolved' | 'all';

export interface WorkRecommendationRead {
  id: string;
  trigger_type: WorkRecommendationTriggerTypeValue;
  trigger_code: string;
  source_signature: string;
  inspection_id: string;
  room_id: string;
  surface_id: string | null;
  target_kind: WorkRecommendationTargetKindValue;
  recommended_work_code: string;
  status: WorkRecommendationStatusValue;
  is_active: boolean;
  resolved_at: string | null;
  accepted_at: string | null;
  dismissed_at: string | null;
  resolved_price_item_id: string | null;
  created_at: string;
  updated_at: string;
  current_price_item: SurfacePriceItemSummaryRead | null;
}

export interface WorkRecommendationListResponse {
  items: WorkRecommendationRead[];
  total: number;
}

export interface WorkRecommendationEvaluateResponse {
  created: number;
  reactivated: number;
  unchanged: number;
  resolved: number;
  items: WorkRecommendationRead[];
  total: number;
}

File: backend/app/schemas/work_recommendation.py
"""Pydantic schemas for Stage 11B.2 recommendation materialization/lifecycle.

Read schemas map from ORM attributes. `current_price_item` is a read-time-only
PriceBook resolution preview computed by the service/endpoint and attached as
a transient attribute — it is never persisted and never touches
`resolved_price_item_id`, which stays reserved as the Stage 11C acceptance
snapshot. `None` means the semantic `recommended_work_code` does not
currently resolve to any owner PriceItem; a non-null value with
`is_archived=True` means it resolves but cannot be newly accepted per Stage
10's existing archived-item rule. Market/reference prices are never
substituted here or anywhere else.
"""
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.work_recommendation import (
    WorkRecommendationStatus,
    WorkRecommendationTargetKind,
    WorkRecommendationTriggerType,
)
from app.schemas.work_plan import SurfacePriceItemSummaryRead


class WorkRecommendationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    trigger_type: WorkRecommendationTriggerType
    trigger_code: str
    source_signature: str
    inspection_id: uuid.UUID
    room_id: uuid.UUID
    surface_id: uuid.UUID | None
    target_kind: WorkRecommendationTargetKind
    recommended_work_code: str
    status: WorkRecommendationStatus
    is_active: bool
    resolved_at: datetime | None
    accepted_at: datetime | None
    dismissed_at: datetime | None
    resolved_price_item_id: uuid.UUID | None
    created_at: datetime
    updated_at: datetime
    current_price_item: SurfacePriceItemSummaryRead | None = None


class WorkRecommendationListResponse(BaseModel):
    items: list[WorkRecommendationRead]
    total: int


class WorkRecommendationEvaluateResponse(BaseModel):
    """Result of one explicit room-wide reconciliation pass.

    `items`/`total` are the room's full current ACTIVE set after
    reconciliation (mirrors RiskDetailResponse's own evaluate-response shape),
    not just the rows touched this call.
    """

    created: int
    reactivated: int
    unchanged: int
    resolved: int
    items: list[WorkRecommendationRead]
    total: int
